### PR TITLE
Redux recompose refactor done

### PR DESCRIPTION
**Pull request by axelfratoni**
Reducers are now written according the redux recompose library.

### CODE REVIEW

File: React/wbooks/src/redux/auth/actions.js [hunk 1]
```diff
@@ -1,49 +1,45 @@
...
-export const authActions = {
-  logout: 'LOGOUT',
-  login: 'LOGIN',
-  loginSuccess: 'LOGIN_SUCCESS',
-  loginFailure: 'LOGIN_FAILURE',
-  signup: 'SIGNUP',
-  signupSuccess: 'SIGNUP_SUCCESS',
-  signupFailure: 'SIGNUP_FAILURE'
-};
+export const actions = createTypes(completeTypes(['LOGOUT', 'LOGIN', 'SIGNUP']), '@@AUTH');
```
Comment (damfinkel) [inline React/wbooks/src/redux/auth/actions.js]: Complete types are those that define a SUCCESS and FAILURE action, which you don't have in logout:
https://github.com/Wolox/redux-recompose/blob/master/src/completers/completeTypes/docs.md

completeTypes receives 2 parameters, the first is an array of thunk actions (those that have success or failure), and the second one is an array of those actions that don't have succes or failure (your logout, for example). So you should be doing the following:

```
export const actions = createTypes(completeTypes(['LOGIN', 'SIGNUP'], ['LOGOUT']), '@@AUTH');
```

File: React/wbooks/src/Services/UserService.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import api from '../Config/api';
 
 const UserService = {
-  login: async (email, password) => api.post('/users/sessions', { email, password }),
+  login: async user => api.post('/users/sessions', { ...user }),
```
Comment (mvbattan) [inline React/wbooks/src/Services/UserService.js]: This shouldn't need to be a copy of `user` (i.e. `...user`), we can pass it directly in the second parameter 😄

File: React/wbooks/src/redux/auth/actions.js [hunk 3]
```diff
@@ -1,49 +1,45 @@
...
 
 const ActionCreators = {
   logout: () => dispatch => {
     localStorage.removeSessionToken();
     api.setHeader('Authorization', '');
     history.push('/login');
```
Comment (mvbattan) [inline React/wbooks/src/redux/auth/actions.js]: This should belong to route constants

File: React/wbooks/src/redux/auth/actions.js [hunk 4]
```diff
@@ -1,49 +1,45 @@
...
+  login: (email, password) => ({
+    type: actions.LOGIN,
+    target: 'login',
+    service: UserService.login,
+    payload: { email, password },
+    injections: [
+      withPostSuccess((dispatch, response) => {
+        localStorage.setSessionToken(response.data.access_token);
+        api.setHeader('Authorization', response.data.access_token);
+        history.push('/');
```
Comment (mvbattan) [inline React/wbooks/src/redux/auth/actions.js]: This `'/'` should belong to route constants

File: React/wbooks/src/redux/auth/actions.js [hunk 5]
```diff
@@ -1,49 +1,45 @@
...
+  signup: user => ({
+    type: actions.SIGNUP,
+    target: 'signup',
+    service: UserService.signup,
+    payload: user,
+    injections: [
+      withPostSuccess((dispatch, response) => {
+        localStorage.setSessionToken(response.data.access_token);
+        api.setHeader('Authorization', response.data.access_token);
+        history.push('/');
```
Comment (mvbattan) [inline React/wbooks/src/redux/auth/actions.js]: This should belong to route constants

File: React/wbooks/src/redux/auth/actions.js [hunk 6]
```diff
@@ -1,49 +1,45 @@
...
-    } else {
-      dispatch({ type: authActions.signupFailure });
-    }
-  }
+  login: (email, password) => ({
+    type: actions.LOGIN,
+    target: 'login',
+    service: UserService.login,
+    payload: { email, password },
+    injections: [
```
Comment (mvbattan) [inline React/wbooks/src/redux/auth/actions.js]: A single injection does not need to be under an array 😄

File: React/wbooks/src/redux/auth/actions.js [hunk 7]
```diff
@@ -1,49 +1,45 @@
...
+      withPostSuccess((dispatch, response) => {
```
Comment (mvbattan) [inline React/wbooks/src/redux/auth/actions.js]: I'd extract this `(dispatch, response)` function [~1]

File: React/wbooks/src/redux/auth/actions.js [hunk 8]
```diff
@@ -1,49 +1,45 @@
...
+      })
+    ]
+  }),
+  signup: user => ({
+    type: actions.SIGNUP,
+    target: 'signup',
+    service: UserService.signup,
+    payload: user,
+    injections: [
+      withPostSuccess((dispatch, response) => {
```
Comment (mvbattan) [inline React/wbooks/src/redux/auth/actions.js]: I'd reuse [~1]

**Review by mvbattan — COMMENTED** [review]
Last comments. Good Job !

**Review by mvbattan — APPROVED** [review]

**Pull request closed without merging**